### PR TITLE
Add spec for results.yaml and report.yaml

### PR DESCRIPTION
**Pull request by thrix**
This provides the results of tests and defines the possible
attributes and expected values.

Signed-off-by: Miroslav Vadkerti <mvadkert@redhat.com>

### CODE REVIEW

File: spec/steps/execute.fmf [hunk 1]
```diff
@@ -13,6 +13,41 @@ description: |
     restraint
         use the restraint harness to execute beakerlib tests
 
+    For each ``plan`` the execute step should produce a ``results.yaml`` file
+    with the list of results for each test in the following format::
+
+        result: PLAN_RESULT
```
Comment (psss) [inline spec/steps/execute.fmf]: At first sight it seemed to me as a good idea but then I realized: Which step should be responsible for aggregating and summarizing the results? Isn't this a task for the `report` step? To look through all test plans, aggregate results of individual tests and show the overall result?

One real life question: If the same test case is executed in multiple plans shall we aggregate the results when reporting to the user? For example running across all supported architectures.

File: spec/steps/execute.fmf [hunk 2]
```diff
@@ -13,6 +13,41 @@ description: |
...
+
+        * info - test finished and produced only information message
+        * passed - test finished and passed
+        * failed - test finished and failed
+        * error - test did not finish, treated as failed
+
+    Note the priority of test results is as written above, with ``info`` having the lowest priority
+    and ``error`` has the highest. This is important for the PLAN_RESULT.
+
+    Where ``PLAN_RESULT`` is the overall result or all test results for the plan run. It has the same
```
Comment (psss) [inline spec/steps/execute.fmf]:
```suggestion
    Where ``PLAN_RESULT`` is the overall result of all test results for the plan run. It has the same
```

File: spec/steps/execute.fmf [hunk 3]
```diff
@@ -13,6 +13,26 @@ description: |
...
+        /test/two:
+            result: TEST_RESULT
+            log: PATH
+
+    Where ``TEST_RESULT`` is the outcome of the test. It can have these values:
+
+        * info - test finished and produced only information message
+        * passed - test finished and passed
+        * failed - test finished and failed
+        * error - test did not finish, treated as failed
```
Comment (psss) [inline spec/steps/execute.fmf]: Test can finish and still errors can happen during execution. What about `a problem encountered during test execution, treated as failed`?

File: spec/steps/report.fmf [hunk 4]
```diff
@@ -1,7 +1,51 @@
 summary: Report test results
 
-description:
-    Adjusting notifications about the test progress and results.
+description: |
+    Create report for all the test results. The report is saved into ``report.yaml``
+    file with the following format::
+
+        /plan/one:
+            result: OVERALL_RESULT
```
[the remark this answers is not in the document]
Comment (thrix) [reply spec/steps/report.fmf]: ty fixing

File: spec/steps/report.fmf [hunk 5]
```diff
@@ -1,7 +1,51 @@
...
+        * error - a problem encountered during test execution
+
+    Note the priority  of test results is as written above, with ``info`` having the lowest priority 
+    and ``error`` has the highest. This is important for ``PLAN_RESULT``.
+
+    Where ``PLAN_RESULT`` is the overall result or all test results for the plan run. It has the same
+    values as ``TEST_RESULT``. Plan result is counted according to the priority of the test outcome
+    values. For example:
+
+        * if the test results are info, passed, passed - the plan result will be passed
```
Comment (lukaszachy) [inline spec/steps/report.fmf]: Dulicate "passed" is intentional?

File: spec/steps/execute.fmf [hunk 6]
```diff
@@ -13,6 +13,26 @@ description: |
...
+            log: PATH
+
+    Where ``TEST_RESULT`` is the outcome of the test. It can have these values:
+
+        * info - test finished and produced only information message
+        * passed - test finished and passed
+        * failed - test finished and failed
+        * error - a problem encountered during test execution, treated as failed
+
+    Where ``PATH`` is the test log output path, relative to the execute step plan run directory.
```
Comment (lukaszachy) [inline spec/steps/execute.fmf]: I take that 'log' is the text output file from the test execution (in beakerlib word the 'testout.log').
What about another artifacts produced by test, which are necessary for failure investigation (in beakerlib anything rlFileSubmit-ed)? Usually stuff as journal, configs, executed commands outputs, coredumps ;)

Comment (psss) [reply spec/steps/execute.fmf]: I think `log` attribute should support both single file and list of files.

File: spec/steps/report.fmf [hunk 7]
```diff
@@ -1,7 +1,51 @@
 summary: Report test results
 
-description:
-    Adjusting notifications about the test progress and results.
+description: |
+    Create report for all the test results. The report is saved into ``report.yaml``
+    file with the following format::
+
+        /plan/one:
```
Comment (thrix) [inline spec/steps/report.fmf]: @psss maybe we should have an overall result also across all plans?

Comment (psss) [reply spec/steps/report.fmf]: Yes, makes sense to me to put a `result` key in the top level of the result structure as the overall result.

File: spec/steps/execute.fmf (hunk 6 above)
Comment (thrix) [reply spec/steps/execute.fmf]: ack, so I will make it a list. @lukaszachy is sufficient just list of paths to logs?

File: spec/steps/report.fmf (hunk 7 above)
Comment (thrix) [reply spec/steps/report.fmf]: ack

File: spec/steps/execute.fmf (hunk 6 above)
Comment (thrix) [reply spec/steps/execute.fmf]: fixed

File: spec/steps/report.fmf (hunk 7 above)
Comment (thrix) [reply spec/steps/report.fmf]: fixed, I also put the plans under `plans:` key, mixing together result and tests on top level feels weird ...

**Comment by thrix** [comment]
@lukaszachy @psss ready for next round of review

**Comment by lukaszachy** [comment]
LGTM

**Comment by psss** [comment]
Good, thanks to all for the review!

**Pull request merged**